.gitignore

```diff
@@ -15,3 +15,4 @@ coverage
 .DS_Store
 .vscode
 *.log
+.*.sw?
```

Comment:
what kind of file is this? is this exclusion a change that you can't just keep on your local machine?

Reply:
vim is a pretty common editor.  When vim opens a file like foo.txt, it creates a temporary file named .foo.txt.swp which holds editor state.  They shouldn't be committed.  Seems to fit with excluding .DS_Store, .vscode, etc.

If you don't want to keep this part, that's okay.

Reply:
Before merging, I'm cool with excluding Vi/Vim/Neovim swap files, but instead of this rule, perhaps we should take the rules from [github/gitignore](https://github.com/github/gitignore/blob/main/Global/Vim.gitignore):

```suggestion

# https://github.com/github/gitignore/blob/main/Global/Vim.gitignore
[._]*.s[a-v][a-z]
!*.svg  # comment out if you don't need vector files
[._]*.sw[a-p]
[._]s[a-rt-v][a-z]
[._]ss[a-gi-z]
[._]sw[a-p]
```

Would you mind doing that as part of this PR?

For future reference, you can take the approach from the comment below too. The feature was actually intended for editor-specific files:

* https://github.com/svg/svgo/pull/1932#discussion_r1445458975

Reply:
I accepted the suggested change.

Yes, I am aware of the .gitignore hierarchy.  My preference is to favor making a project self-contained for really common bits to make it harder for people, especially occasional committers, to accidentally commit incorrect stuff.  But also, I am a rare committer on this project and yield 100% to the maintainers on this point.

No worries on the rebasing.  Not even completely sure it was the rebasing that broke it.  But we have a fix and a test now!  Thanks for the quick review!